File: modules/app/cli/src/main/java/wtf/metio/ilo/cli/Ilo.java
/*
 * This file is part of ilo. It is subject to the license terms in the LICENSE file found in the top-level
 * directory of this distribution and at http://creativecommons.org/publicdomain/zero/1.0/. No part of ilo,
 * including this file, may be copied, modified, propagated, or distributed except according to the terms contained
 * in the LICENSE file.
 *
 */

package wtf.metio.ilo.cli;

import picocli.AutoComplete;
import picocli.CommandLine;
import wtf.metio.ilo.cli.commands.OpenShell;
import wtf.metio.ilo.cli.options.ShellOptions;

import java.util.Arrays;
import java.util.concurrent.Callable;

/**
 * Main entry point for Ilo - the reproducible build environment manager
 */
@CommandLine.Command(
    name = "ilo",
    description = "Manage reproducible build environments",
    version = "1.0.0",
    mixinStandardHelpOptions = true,
    usageHelpAutoWidth = true,
    subcommands = {
        OpenShell.class,
        AutoComplete.GenerateCompletion.class
    },
    showDefaultValues = true
)
public final class Ilo implements Callable<Integer> {

  @CommandLine.Mixin
  private ShellOptions options;

  public static void main(final String[] args) {
    // TODO: parse .ilo.rc
    System.exit(new CommandLine(new Ilo())
        .execute(Arrays.stream(args)
            .filter(arg -> !Ilo.class.getCanonicalName().equalsIgnoreCase(arg))
            .toArray(String[]::new)));
  }

  @Override
  public Integer call() {
    return new OpenShell(options).call();
  }

}

File: HomebrewFormula/ilo.rb
# This file is part of ilo. It is subject to the license terms in the LICENSE file found in the top-level
# directory of this distribution and at https://creativecommons.org/publicdomain/zero/1.0/. No part of ilo,
# including this file, may be copied, modified, propagated, or distributed except according to the terms contained
# in the LICENSE file.

class Ilo < Formula
  desc "Manage reproducible build environments"
  version "2023.12.4"
  homepage "https://ilo.projects.metio.wtf/"
  license "CC0"
  url "https://github.com/metio/ilo/releases/download/2023.12.4/ilo-2023.12.4-mac.zip"
  sha256 "fff4e378271c6239934575e4fe9ce74128303fb2d6dabee0b7d230997b9b6ac4"

  depends_on "podman" => :optional
  depends_on "docker" => :optional
  depends_on "nerdctl" => :optional

  def install
    bin.install "ilo"
  end

  test do
    system "#{bin}/ilo", "--help"
  end
end
